stetho/src/main/java/com/facebook/stetho/inspector/network/AsyncPrettyPrinterSchemaFileManager.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.stetho.inspector.network;
+
+import android.content.Context;
+import com.facebook.stetho.common.LogRedirector;
+import com.facebook.stetho.common.Util;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URL;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class AsyncPrettyPrinterSchemaFileManager {
+  private static final String TAG = "AsyncPrettyPrinterSchemaCacheManager";
+  private static final String FILENAME_PREFIX = "async-pretty-printer-schema-";
+  private static AtomicInteger sCacheFileCounter = new AtomicInteger(0);
+  private final Map<URL, String> mFilenameMap = Collections.synchronizedMap(
+      new HashMap<URL,String>());
+
+  private final Context mContext;
+
+  public AsyncPrettyPrinterSchemaFileManager(Context context) {
+    mContext = context;
+  }
+
+  public OutputStream openNewSchemaCache(URL schemaUrl)
+      throws IOException {
+    synchronized (mFilenameMap) {
```

Comment:
We don't need the `synchronized` anymore: `sCacheFileCounter` is atomic, `mFilenameMap` is already synchronized, and other than that the remaining statements don't appear to need group atomicity.

However .............. what happens if `schemaUrl` is already present in the `mFilenameMap` ? that seems like an error, right? If so, then we should 1) not bother with `AtomicInteger` for `sCacheFileCounter`, 2) not bother with making `mFilenameMap` synchronized, and 3) do our own synchronization (yeah that can mean making the methods synchronized again :-) ), and 4) check if `mFilenameMap.containsKey(schemaUrl)` before blindly putting it in there